### PR TITLE
Make end portal generation drop blocks and refactor code 

### LINKED ISSUE #266: The obsidian platform in the end is generated in wrong y places for MC 1.21
### Information

* **Server version:** This server is running CraftBukkit version 4242-Spigot-0a642bd-515fe49 (MC: 1.21) (Implementing API version 1.21-R0.1-SNAPSHOT)

* **Full output of `/mv version -p`:** https://paste.gg/p/anonymous/701c684147d64c7c9492ba7e1d3a51e3

* **Server log:** https://gist.github.com/KagefumiMerry/2e8384a4a70751bacace0e7f44012dc0

### Details
I was **able** to reproduce my issue on a freshly setup and up-to-date server with the latest version of Multiverse plugins with no other plugins and with no kinds of other server or client mods.

**Description**  
When player enter the end, the obsidian platform is generated wrongly above the platform generated by the Multiverse-NetherPortals.
Once Multiverse-NetherPortals plugin removed, the behavior returned to vanilla one.

**Steps to reproduce**  
1. Enter the end portal in the stronghold 
2. the obsidian platform is generated, but with double its thickness.
3. all the blocks not obsidian or air are just missing

**Expected behavior**  
1. the obsidian platform should be generated
2. all the blocks not obsidian or air should be the drop item entities (the 1.21 behavior)

**Screenshots**  

I check a bit the source code, it maybe related the NetherPortals handling in
src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java
src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPPlayerListener.java

```
            } else if (toWorld.getEnvironment() == World.Environment.THE_END && type == PortalType.ENDER) {
                Location loc = new Location(event.getTo().getWorld(), 100, 50, 0); // This is the vanilla location for obsidian platform.
                event.setTo(loc);
                Block block = loc.getBlock();
                for (int x = block.getX() - 2; x <= block.getX() + 2; x++) {
                    for (int z = block.getZ() - 2; z <= block.getZ() + 2; z++) {
                        Block platformBlock = loc.getWorld().getBlockAt(x, block.getY() - 1, z);
                        if (platformBlock.getType() != Material.OBSIDIAN) {
                            platformBlock.setType(Material.OBSIDIAN);
                        }
                        for (int yMod = 1; yMod <= 3; yMod++) {
                            Block b = platformBlock.getRelative(BlockFace.UP, yMod);
                            if (b.getType() != Material.AIR) {
                                b.setType(Material.AIR);
                            }
                        }
                    }
                }
            }
```

Also, I need to mention that Entity spawn in a bit different position, but current spawn point command only support single type.

From Wiki:
Players who enter the End spawn at coordinates (100, 49, 0), facing west towards the main island; other entities that enter the End spawn at coordinates (100.5, 50, 0.5), in the middle of and 1 block higher than the platform

### PR DESCRIPTION
**Pull request by zax71**
Fixes #266 

Aligns behaviour of that introduced in 1.21, also some code refactoring was done

### CODE REVIEW

File: src/main/java/com/onarandombox/MultiverseNetherPortals/utils/EndPlatformCreator.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+public class EndPlatformCreator {
+
+    /**
+     * Creates an end platform at the specified {@code Block}
+     * @param spawnLocation The {@code Block} that the player will spawn at.
+     */
+    public static void createEndPlatform(Block spawnLocation) {
+        Logging.fine("Creating an end platform at " + spawnLocation.toString());
+
+        // Is this code even required?! - Past self
```
Comment (benwoo1110) [inline src/main/java/com/onarandombox/MultiverseNetherPortals/utils/EndPlatformCreator.java:20]: Hm?

File: src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java [hunk 2]
```diff
@@ -299,14 +299,25 @@ public void onEntityPortal(EntityPortalEvent event) {
...
         }
 
-        Entity e = event.getEntity();
-        Location originalTo = this.locationManipulation.getBlockLocation(event.getTo());
-        Location currentLocation = this.locationManipulation.getBlockLocation(event.getFrom());
+        // Some shortcuts for later
+        Entity entity = event.getEntity();
 
+        Location toLocation = event.getTo();
+        Location fromLocation = event.getFrom();
```
Comment (benwoo1110) [inline src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java:306]: Is it possible for either of this to be nullable? Need to check the annotation

**Review by benwoo1110 — COMMENTED** [review]
Uhm which line exactly fixes the block teleport issue?

**Comment by zax71** [comment]
> Uhm which line exactly fixes the block issue?

`b.breakNaturally();`

File: src/main/java/com/onarandombox/MultiverseNetherPortals/utils/EndPlatformCreator.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+     */
+    public static void createEndPlatform(Location spawnLocation) {
+        createEndPlatform(spawnLocation.getBlock());
+    }
+
+    /**
+     * The default vanilla location for the end platform
+     */
+    public static Location getVanillaLocation(World world) {
+        return new Location(world, 100, 49, 0, 90, 0);
```
[the remark this answers is not in the document]
Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseNetherPortals/utils/EndPlatformCreator.java:57]: Yep

File: src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java [hunk 4]
```diff
@@ -343,57 +355,53 @@ public void onEntityPortal(EntityPortalEvent event) {
...
-        if (!event.isCancelled()) {
-            if (fromWorld.getEnvironment() == World.Environment.THE_END && type == PortalType.ENDER) {
-                Logging.fine("Entity '" + e.getName() + "' will be teleported to the spawn of '" + toWorld.getName() + "' since they used an end exit portal.");
-                try {
-                    Class.forName("org.bukkit.TravelAgent");
-                    event.getPortalTravelAgent().setCanCreatePortal(false);
-                } catch (ClassNotFoundException ignore) {
-                    Logging.fine("TravelAgent not available for EntityPortalEvent for " + e.getName() + ". There may be a portal created at spawn.");
-                }
+        if (event.isCancelled()) { // Superfluous?
```
[the remark this answers is not in the document]
Comment (zax71) [reply src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java:358]: Precisely, it was there before so I left it in, was unsure if you'd know why it was there

[the remark this answers is not in the document]
Comment (benwoo1110) [reply src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java:358]: I think let's remove it

**Comment by zax71** [comment]
Valid, a new config option like "end_platform_drop_blocks"?

**Review by benwoo1110 — APPROVED** [review]

**Comment by zax71** [comment]
🎉

**Pull request merged**